src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -450,36 +494,36 @@ export default {
         // Effective Date may be in the future (eg, for Incorp App future effective filings).
         // If Effective Date is empty, use Filing Date instead.
         const effectiveDateTime = this.convertUTCTimeToLocalTime(header.effectiveDate) || filingDateTime
-        const effectiveDate = effectiveDateTime?.slice(0, 10)
 
         // is this a Future Effective Incorp App?
-        const isIaFutureEffective = this.isFutureEffective(filing)
-
-        // is this a paid filing?
-        const isPaid = (header.status === FilingStatus.PAID)
+        const isFutureEffectiveIa = !!filing.header.isFutureEffective
 
-        // is this a completed filing?
-        const isCompleteIA = (header.status === FilingStatus.COMPLETED)
+        // is this a Future Effective Incorp App pending completion?
+        const isFutureEffectiveIaPending = isFutureEffectiveIa && this.isEffectiveDatePast(filing)
 
         const name = this.filingTypeToName(filingType)
+
         let receiptFilename: string
-        if (isIaFutureEffective) {
+        if (isFutureEffectiveIa) {
           receiptFilename = `${this.entityIncNo} - Receipt (Future Effective) - ${filingDate}.pdf`
         } else {
           receiptFilename = `${this.entityIncNo} - Receipt - ${filingDate}.pdf`
         }
 
+        const corpName = this.entityName || this.entityTypeToTitle(this.entityType)
```

Comment:
Nice clean up, this essentially a fix to that `Numbered Benefit Company` bug as well.

Reply:
Yup, thanks.

I tried to clean this up so it's easier to understand and extend in the future :)